src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/TypeGVMEntriesNode.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.TypeSystem;
+using ILCompiler.DependencyAnalysisFramework;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// This node is used for GVM dependency analysis and GVM tables building. Given an input
+    /// type, this node can scan the type for a list of GVM table entries, and compute their dependencies.
+    /// </summary>
+    internal sealed class TypeGVMEntriesNode : DependencyNodeCore<NodeFactory>
+    {
+        private TypeDesc _associatedType;
+        private DependencyList _staticDependencies;
+
+        public TypeGVMEntriesNode(TypeDesc associatedType)
+        {
+            Debug.Assert(!associatedType.IsRuntimeDeterminedSubtype);
+            Debug.Assert(TypeNeedsGVMTableEntries(associatedType));
+            _associatedType = associatedType;
+        }
+
+        public override bool HasConditionalStaticDependencies => false;
+        public override bool HasDynamicDependencies => false;
+        public override bool InterestingForDynamicDependencyAnalysis => false;
+        public override bool StaticDependenciesAreComputed => true;
+        protected override string GetName() => null;
+        public override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDependencies(NodeFactory context)
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+        public override IEnumerable<CombinedDependencyListEntry> SearchDynamicDependencies(List<DependencyNodeCore<NodeFactory>> markedNodes, int firstNode, NodeFactory context)
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+
+        public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFactory context)
+        {
+            if (_staticDependencies == null)
+            {
+                _staticDependencies = new DependencyList();
+
+                ScanForGenericVirtualMethodEntries(context, TrackGVMDependencies);
+                ScanForInterfaceGenericVirtualMethodEntries(context, TrackInterfaceGVMDependencies);
+            }
+
+            return _staticDependencies;
+        }
+
+        public static bool TypeNeedsGVMTableEntries(TypeDesc type)
+        {
+            if (type.Category != TypeFlags.Class && !type.IsValueType)
```

Comment:
Using .IsDefType is more readable I agree :). I just didn't know about it before

---
In reply to: [96550314](https://github.com/dotnet/corert/pull/2521#discussion_r96550314) [](ancestors = 96550314)